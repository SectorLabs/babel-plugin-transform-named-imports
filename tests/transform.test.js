const path = require('path');
const pluginTester = require('babel-plugin-tester');
const createBabylonOptions = require('babylon-options');

const plugin = require('../src/index.js');

const babelOptions = {
    filename: 'currentFile.js',
    parserOpts: createBabylonOptions({
        stage: 2
    })
};

pluginTester({
    plugin,
    babelOptions: babelOptions,
    pluginOptions: {
        webpackConfig: path.resolve(__dirname + '/webpack.config.js'),
    },
    snapshot: true,
    tests: {
        // convert this into a default import that leads to `testmodule/myFunc`
        'single named import':
            `import { Foo2 } from 'testmodule'`,

        // convert this into named import that leads to `testmodule/myOtherFunc`
        'nested named import':
            `import { myOtherFunc } from 'testmodule'`,

        // convert this into one default import that leads to `testmodule/myFunc`
        // and a named import that leads to `testmodule/myOtherFunc`
        'multiple named imports':
            `import { myFunc, myOtherFunc } from 'testmodule'`,

        // convert this into three imports, one default import for `init` and
        // one default import for `myFunc` and a named one for `myOtherFunc`
        'default import with multiple named import':
            `import init, { myFunc, myOtherFunc } from 'testmodule'`,

        // convert this into a default import with `myAliasFunc` leading to
        // `testmodule/myFunc`
        'aliased named import':
            `import { myFunc as myAliasFunc } from 'testmodule'`,

        // don't change existing default imports like this
        'default import':
            `import myFunc from 'testmodule/myFunc'`,

        // unresolved default imports should be left alone
        'unresolved default import':
            `import React from 'reacty'`,

        // unresolved imports should be left alone
        'unresolved default import with named import':
            `import React, { Component } from 'reacty'`,

        // common js imports should be left alone
        'common js default import':
            `import React from './commonjsmodule'`,

        // common js imports should be left alone
        'common js default with named import':
            `import React, { Component } from './commonjsmodule'`,

        // convert this into a namespace import that leads to `testmodule/constants`
        'aliased namespace import':
            `import { thangs } from 'testmodule'`,

        // leaves glob imports alone
        'glob import':
            `import * as testmodule from 'testmodule'`,
    },
});
