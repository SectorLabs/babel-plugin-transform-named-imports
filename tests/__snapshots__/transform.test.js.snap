// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform-named-imports aliased named import: aliased named import 1`] = `
"
import { myFunc as myAliasFunc } from 'testmodule'

      ↓ ↓ ↓ ↓ ↓ ↓

import myAliasFunc from \\"./tests/testmodule/myFunc.js\\";
"
`;

exports[`transform-named-imports aliased namespace import: aliased namespace import 1`] = `
"
import { thangs } from 'testmodule'

      ↓ ↓ ↓ ↓ ↓ ↓

import * as thangs from \\"./tests/testmodule/constants.js\\";
"
`;

exports[`transform-named-imports common js default import: common js default import 1`] = `
"
import React from './commonjsmodule'

      ↓ ↓ ↓ ↓ ↓ ↓

import React from './commonjsmodule';
"
`;

exports[`transform-named-imports common js default with named import: common js default with named import 1`] = `
"
import React, { Component } from './commonjsmodule'

      ↓ ↓ ↓ ↓ ↓ ↓

import React, { Component } from './commonjsmodule';
"
`;

exports[`transform-named-imports default import with multiple named import: default import with multiple named import 1`] = `
"
import init, { myFunc, myOtherFunc } from 'testmodule'

      ↓ ↓ ↓ ↓ ↓ ↓

import init from \\"./tests/testmodule/index.js\\";
import myFunc from \\"./tests/testmodule/myFunc.js\\";
import { myOtherFunc } from \\"./tests/testmodule/myOtherFunc.js\\";
"
`;

exports[`transform-named-imports default import: default import 1`] = `
"
import myFunc from 'testmodule/myFunc'

      ↓ ↓ ↓ ↓ ↓ ↓

import myFunc from 'testmodule/myFunc';
"
`;

exports[`transform-named-imports glob import: glob import 1`] = `
"
import * as testmodule from 'testmodule'

      ↓ ↓ ↓ ↓ ↓ ↓

import * as testmodule from 'testmodule';
"
`;

exports[`transform-named-imports import export in single line: import export in single line 1`] = `
"
import { myInlineExport } from 'testmodule'

      ↓ ↓ ↓ ↓ ↓ ↓

import { myInlineFunc as myInlineExport } from \\"./tests/testmodule/myOtherFunc.js\\";
"
`;

exports[`transform-named-imports multiple named imports: multiple named imports 1`] = `
"
import { myFunc, myOtherFunc } from 'testmodule'

      ↓ ↓ ↓ ↓ ↓ ↓

import myFunc from \\"./tests/testmodule/myFunc.js\\";
import { myOtherFunc } from \\"./tests/testmodule/myOtherFunc.js\\";
"
`;

exports[`transform-named-imports nested named import: nested named import 1`] = `
"
import { myOtherFunc } from 'testmodule'

      ↓ ↓ ↓ ↓ ↓ ↓

import { myOtherFunc } from \\"./tests/testmodule/myOtherFunc.js\\";
"
`;

exports[`transform-named-imports single named import: single named import 1`] = `
"
import { myFunc } from 'testmodule'

      ↓ ↓ ↓ ↓ ↓ ↓

import myFunc from \\"./tests/testmodule/myFunc.js\\";
"
`;

exports[`transform-named-imports unresolved default import with named import: unresolved default import with named import 1`] = `
"
import React, { Component } from 'reacty'

      ↓ ↓ ↓ ↓ ↓ ↓

import React, { Component } from 'reacty';
"
`;

exports[`transform-named-imports unresolved default import: unresolved default import 1`] = `
"
import React from 'reacty'

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'reacty';
"
`;
